test/xdotoolify-spec.js

```diff
@@ -493,6 +509,76 @@ describe('xdotoolify', function() {
       'Most recent value: {"a":1,"b":2}\n' +
       'Most recent check result: {"a":1,"b":2}\n'
     );
+  });
+
+  it('should accept Xdotoolify.arg as argument', async function() {
+    let goodFunc = Xdotoolify.setupWithPage(async (page) => { console.log('running'); return {
+      a: 1,
+      b: 2
+    }; });
+
+    let funcWithArgs = Xdotoolify.setupWithPage((page, arg) => { return arg; });
+
+    let errorMsg = 'Nothing thrown';
+
+    try {
+      await page.X
+          .checkUntil(
+            funcWithArgs,
+            Xdotoolify.arg(async (_page) => {
+              return (await goodFunc(page)).a;
+            }, page),
+            x => x,
+            1
+          )
+          .do()
+    } catch (e) {
+      errorMsg = e.message;
+    }
+
+    expect(errorMsg).toBe('Nothing thrown');
+
+    try {
+      await page.X
+          .checkUntil(
+            funcWithArgs,
+            Xdotoolify.arg(() => {
+              return 3;
+            }),
+            x => x,
+            3
+          )
+          .do()
+    } catch (e) {
+      errorMsg = e.message;
+    }
 
+    expect(errorMsg).toBe('Nothing thrown');
+  });
+
+  it('should accept save values and retrieve them', async function() {
+    let funcWithArgs = Xdotoolify.setupWithPage((page, arg) => { return arg; });
+
+    let errorMsg = 'Nothing thrown';
+
+    try {
+      await page.X
+          .saveValue(
+            funcWithArgs,
+            'result',
```

Comment:
I think we found a better api than this. The problem with this is it does not allow you to do any sort of calculation before saving. Better would be as follows:

```
let value = null;
await page.X
    .run(Xdotoolify.setupWithoutPage(() => {
      value = someTransformation(await someApi.call());
    }))
    .run(submitValue, defer(() => value)))
    .checkUntil(myFunc, x => expect(x).toBe(value))
    .do();
```

Reply:
I don't think we should have the ability to `getSavedValue` outside of a chain. i.e. `defer` would be the only way to get it.

Also, Xdotoolify.verify should ensure that there are no dangling `defer`s that were not used (since that would indicate the Xdotoolify api was misused)

Reply:
Removed the code related to saving, and added a test for the procedure you described.